### PR TITLE
Introduce LinkedIn OIDC provider

### LINKED ISSUE #37726: Support OIDC LinkedIn provider
### Description

Provide `quarkus.oidc.provider=linkedin` option to make it easy to work with LinkedIn

### Implementation ideas

_No response_

### PR DESCRIPTION
**Pull request by gastaldi**
It works!

```properties
quarkus.oidc.provider=linkedin
quarkus.oidc.client-id=<CLIENT_ID>
quarkus.oidc.credentials.client-secret.value=<CLIENT_SECRET>
```

- Fixes #37726

### CODE REVIEW

**Comment by gastaldi** [comment]
@sberyozkin sorry, I know the issue is assigned to you, but I was curious on seeing it in action and spent some hours trying to make it work :)

**Comment by sberyozkin** [comment]
Hey @gastaldi, thanks a million for giving it a try, appreciated. Nice space optimization in the catch block too :+1:  

I knew this integration had to be done at some point due to the earlier commitment which is why I opened this issue but I only had some images for the docs prepared :-), I'll push some docs update a bit later on. 

FYI, I believe you can optimize since they support OIDC discovery, see https://www.linkedin.com/oauth/.well-known/openid-configuration, so let me suggest some minor updates

File: extensions/oidc/runtime/src/main/java/io/quarkus/oidc/runtime/providers/KnownOidcProviders.java [hunk 1]
```diff
@@ -10,31 +10,34 @@
...
+            case TWITCH -> twitch();
+            case TWITTER, X -> twitter();
+        };
+    }
+
+    private static OidcTenantConfig linkedIn() {
+        OidcTenantConfig ret = new OidcTenantConfig();
+        ret.setAuthServerUrl("https://www.linkedin.com/oauth/v2");
+        ret.setApplicationType(OidcTenantConfig.ApplicationType.WEB_APP);
+        ret.setDiscoveryEnabled(false);
```
Comment (sberyozkin) [inline extensions/oidc/runtime/src/main/java/io/quarkus/oidc/runtime/providers/KnownOidcProviders.java:32]:
```suggestion
```

File: extensions/oidc/runtime/src/main/java/io/quarkus/oidc/runtime/providers/KnownOidcProviders.java [hunk 2]
```diff
@@ -10,31 +10,34 @@
...
+        ret.setAuthorizationPath("authorization");
```
Comment (sberyozkin) [inline extensions/oidc/runtime/src/main/java/io/quarkus/oidc/runtime/providers/KnownOidcProviders.java:33]:
```suggestion
```

File: extensions/oidc/runtime/src/main/java/io/quarkus/oidc/runtime/providers/KnownOidcProviders.java [hunk 3]
```diff
@@ -10,31 +10,34 @@
...
+        ret.setTokenPath("accessToken");
```
Comment (sberyozkin) [inline extensions/oidc/runtime/src/main/java/io/quarkus/oidc/runtime/providers/KnownOidcProviders.java:34]:
```suggestion
```

File: extensions/oidc/runtime/src/main/java/io/quarkus/oidc/runtime/providers/KnownOidcProviders.java [hunk 4]
```diff
@@ -10,31 +10,34 @@
...
+        ret.setUserInfoPath("https://api.linkedin.com/v2/userinfo");
```
Comment (sberyozkin) [inline extensions/oidc/runtime/src/main/java/io/quarkus/oidc/runtime/providers/KnownOidcProviders.java:35]:
```suggestion
```

File: extensions/oidc/runtime/src/main/java/io/quarkus/oidc/runtime/providers/KnownOidcProviders.java [hunk 5]
```diff
@@ -10,31 +10,34 @@
...
+        ret.setJwksPath("https://www.linkedin.com/oauth/openid/jwks");
```
Comment (sberyozkin) [inline extensions/oidc/runtime/src/main/java/io/quarkus/oidc/runtime/providers/KnownOidcProviders.java:36]:
```suggestion
```

**Comment by sberyozkin** [comment]
@gastaldi Hi George, I suggested to remove a few configuration lines, if you'd like, please ignore the, and remove all those line as a single block. Please test the proposed config update when you will have time. Please also add a pair of tests in `OidcUtilsTest` (copying 2 Google tests there would probably be easiest for this case since LinkedIn supports OIDC). I can follow up then with the doc update

**Comment by gastaldi** [comment]
@sberyozkin thank you! I just tested locally and your changes worked like a charm! I'll add some tests and ping you again when it's ready

**Comment by sberyozkin** [comment]
Just had to step outside, thanks @gastaldi for taking care of all of it, incl the docs, cheers

**Review by sberyozkin — APPROVED** [review]

**Pull request merged**